app/controllers/concerns/remember_device_concern.rb

```diff
@@ -12,8 +12,12 @@ def save_remember_device_preference
   def check_remember_device_preference
     return unless authentication_context?
     return if remember_device_cookie.nil?
-    return unless remember_device_cookie.valid_for_user?(current_user)
-    handle_valid_otp
+    return unless remember_device_cookie.valid_for_user?(
+      current_user,
+      expiration_interval: decorated_session.mfa_expiration_interval
+    )
+
+    handle_valid_remember_device_cookie
```

Comment:
This was changed since now `handle_valid_otp` removes the `mfa_device_remembered` flag.